### ISSUE TITLE
Walkthrough not up to date, doesn't work

### ISSUE THREAD

**Issue by srudin** [Azure/aks-engine#1258]
**Is this a request for help?**: Yes

---

**Is this an ISSUE or FEATURE REQUEST?** (choose one): ISSUE

---

**What version of aks-engine?**: Latest

---

**Kubernetes version**: 1.14

**What happened**: There is a recent blog post saying that Kubernetes now supports Windows containers: https://cloudblogs.microsoft.com/opensource/2019/03/25/windows-server-containers-now-supported-kubernetes/
Indeed the documentation here was updated a few days after the post: https://github.com/Azure/aks-engine/blob/master/docs/topics/windows.md
However the examples that are used inside the tutorial have not been updated for a while: https://github.com/Azure/aks-engine/tree/master/examples/windows
Also the yaml example in the tutorial still referrs to "beta.kubernetes.io/os" which feels wrong.
I tried to follow the tutorial but couldn't make it work. The way I see it the whole thing cannot work as long as the yaml examples are not updated. In fact I wonder if the tutorial is reflecting the latest changes at all as it still feels very comlicated to get Windows containers to run. I mean it was already possible to get Windows container to run on Kubernetes before the latest update, but it involved a couple of tweaks (https://anthonychu.ca/post/hybrid-kubernetes-linux-windows-cluster-easy-steps/). I was hoping that everything would work more "out of the box" now.

**What you expected to happen**:
Please verify and update all required resources for the tutorial. I would expect that I can follow and finish the tutorial without running into errors at some point in the middle.

**How to reproduce it** (as minimally and precisely as possible):
Follow the tutorial...

**Anything else we need to know**:
Thank you.

**Comment by munishm** [comment]
The beta in example is not outdated, it is yet to graduate. Please check this- 
https://github.com/kubernetes/enhancements/blob/master/keps/sig-windows/20190103-windows-node-support.md

**Comment by munishm** [comment]
@srudin - Can you please let us know what did not work in the example?

**Comment by srudin** [comment]
@munishm Thanks for the response. I got stuck after generating the template, when trying to deploy it (Deploy the cluster). I don't recall the error and I didn't investigate too much because I was confused that the samples were several months old and I assumed that they are outdated.

Meanwhile I could basically get it to run using another tutorial. Still there is one thing I don't understand: Kubernetes v1.14 is not yet available on Azure. I started this issue because of the blog post announcing windows support out of the box (first link on top). What I'd like to know is what this announement actually relates to - I mean it was possible for some time already to get windows containers to run on Kubernetes and your sample based on a pre-1.14 version shows that. I'd like to know what v1.14 actually helps about that? What does it simplify? I was hoping that all these long and sometimes complicated tutorials where additional functionality has to be seperately activated would become obsolete. Yes, I was hoping that with the "az aks create" command I could pass a parameter for the required os' and - finished. Will this be possible after v1.14 is available? Or what will be the benefit?

**Comment by munishm** [comment]
@srudin - Thank you for your reply. There might be an error in case dns-prefix is same as used by someone else. This is a global resource and might run into allocating the IP to dns. You can try again with a different dns-prefix and see if there is any issue. 

K8s 1.14 has production level windows support now, graduate from beta to stable as stated in the link above. 

cc @craiglpeters @PatrickLang

**Comment by srudin** [comment]
The problem was that I added --no-ssh-keys to the az create command. I figured I wouldn't need to connect to the master and so I could spare generating the keys. Turns out that the whole cluster installation basically works - BUT enabling the virtual-node addon then has no ssh key in the linux profile value and cannot delegate access to that node. Not that I fully understand it but as a matter of fact the problem was solved after passing in an ssh key instead of setting the "ignore" option.
You might say this was my mistake however I find it misleading that there is such an option at all if it leads to unpredictable internal errors. I was only able to find out the problem after comparing a working and a non-working instance step by step.
